core/src/main/java/org/elasticsearch/common/path/PathTrie.java

```diff
@@ -111,7 +113,10 @@ public synchronized void insert(String[] path, int index, T value) {
                 // in case the target(last) node already exist but without a value
                 // than the value should be updated.
                 if (index == (path.length - 1)) {
-                    assert (node.value == null || node.value == value);
+                    if (node.value != null) {
+                        throw new IllegalArgumentException("Path [" + Strings.join(path, "/")+ "] already has a value ["
```

Comment:
Since we lost the "no duplicate" behavior from `ActionModule` I figured we should duplicate it here, but with paths.